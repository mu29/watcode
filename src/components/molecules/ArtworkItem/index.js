import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { Heading, Tag, Text } from 'components'
import { palette } from 'services/style'

const Wrapper = styled.div`
  display: flex;
  padding: 1rem;
  border: 0.0625rem solid ${palette('gray.40')};
  background-color: ${palette('white.default')};
  cursor: pointer;
  transition: background-color 0.2s, box-shadow 0.2s;

  @media(max-width: 639px) {
    flex-direction: column;
  }

  &:hover {
    background-color: ${palette('gray.10')};
    box-shadow: 0 0.125rem 0.125rem 0 rgba(0, 0, 0, 0.1);
  }
`

const HeadingArea = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
  width: 10rem;

  @media(max-width: 639px) {
    width: 100%;
  }
`

const InfoArea = styled.div`
  flex: 1;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  overflow: hidden;
`

const Divider = styled.div`
  width: 0.0625rem;
  margin: 0 1rem;
  background-color: ${palette('gray.30')};

  @media(max-width: 639px) {
    width: 100%;
    height: 0.0625rem;
    margin: 1rem 0;
  }
`

const Title = styled(Heading)`
  width: 100%;
  margin-bottom: 0.5rem;
  overflow: scroll;
  white-space: nowrap;

  ::-webkit-scrollbar {
    width: 0;
    height: 0;
  }

  @media(max-width: 639px) {
    text-align: center;
    margin-bottom: 0.5rem;
  }
`

const TagArea = styled.div`
  width: 100%;
  white-space: nowrap;
  overflow: scroll;

  ::-webkit-scrollbar {
    width: 0;
    height: 0;
  }
`

const TypeTag = styled(Tag)`
  font-weight: 400;
  color: ${palette('white.default')};
  background-color: ${palette('gray.90')};
`

const Image = styled.img`
  height: 5.75rem;
  margin: -1rem;
  margin-left: auto;
`

const ArtworkItem = ({
  artwork: {
    code,
    artist,
    title,
    type,
    tags,
    imageUrl,
  },
  showImage,
  ...props
}) => (
  <Wrapper { ...props }>
    <HeadingArea>
      <Heading level={ 3 }>#{ code }</Heading>
      <Text color="gray.50" fontSize={ 12 } fontWeight={ 200 } small>by { artist }</Text>
    </HeadingArea>
    <Divider />
    <InfoArea>
      <Title level={ 5 } fontWeight={ 400 }>{ title }</Title>
      <TagArea>
        <TypeTag>{ type }</TypeTag>
        { tags.map(tag => (<Tag key={ tag }>{ tag }</Tag>)) }
      </TagArea>
    </InfoArea>
    { showImage && <Image src={ imageUrl } alt="" /> }
  </Wrapper>
)

ArtworkItem.propTypes = {
  artwork: PropTypes.shape({
    code: PropTypes.number,
    artist: PropTypes.string,
    title: PropTypes.string,
    tags: PropTypes.array,
    imageUrl: PropTypes.string,
  }).isRequired,
  showImage: PropTypes.bool,
}

ArtworkItem.defaultProps = {
  showImage: false,
}

export default ArtworkItem
