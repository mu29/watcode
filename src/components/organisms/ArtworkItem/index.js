import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { Heading, Tag, Text, Link, ArtworkImage } from 'components'
import { palette } from 'services/style'

const Wrapper = styled.div`
  display: flex;
  padding: 1rem;
  background-color: ${palette('white.default')};
  border-bottom: 0.0625rem solid ${palette('gray.30')};
`

const InfoArea = styled.div`
  flex: 1;
  height: 7rem;
  overflow: hidden;
`

const TagArea = styled.div`
  height: 3.5rem;
  overflow: scroll;
  margin-top: 0.3125rem;
`

const StyledHeading = styled(Heading)`
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
`

const ArtworkItem = ({
  artwork: {
    code,
    artist,
    title,
    type,
    tags,
    imageUrl,
  },
  showImage,
  ...props
}) => (
  <Wrapper { ...props }>
    <ArtworkImage code={ code } url={ imageUrl } />
    <InfoArea>
      <Link to={ `/artworks/${code}` }>
        <StyledHeading level={ 5 }>{ title }</StyledHeading>
      </Link>
      <Text color="gray.60" fontSize={ 12 } fontWeight={ 200 } small>by { artist }</Text>
      <TagArea>
        <Tag type>{ type }</Tag>
        { tags.map(tag => (<Tag key={ tag }>{ tag }</Tag>)) }
      </TagArea>
    </InfoArea>
  </Wrapper>
)

ArtworkItem.propTypes = {
  artwork: PropTypes.shape({
    code: PropTypes.number,
    artist: PropTypes.string,
    title: PropTypes.string,
    tags: PropTypes.array,
    imageUrl: PropTypes.string,
  }).isRequired,
  showImage: PropTypes.bool,
}

ArtworkItem.defaultProps = {
  showImage: false,
}

export default ArtworkItem
