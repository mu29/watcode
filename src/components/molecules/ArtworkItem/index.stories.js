import React from 'react'
import { storiesOf } from '@storybook/react'
import ArtworkItem from '.'

const artwork = {
  code: 112047,
  artist: 'nagatsuki taffee',
  title: 'Re: 제로부터 시작하는 이세계 생활',
  type: 'Light Novel',
  tags: ['rebirth', 'loop', 'harem', 'this is a long tag'],
  imageUrl: 'https://cdn.namuwikiusercontent.com/s/3f75b87a3b69db1af44479a12315ed58c7e6121c2842cf929209cbf74fa23b7b214a9124e7e494f6ffc8312bdcf231d53278bdf5e7330fb8b23c59ac2ebaf299a3775c6479795372d4b4be0533d58d63?e=1535506665&k=FZxNDPx_iAUdg9ZFqsqXGg',
}

storiesOf('Molecules/ArtworkItem', module)
  .add('기본', () => (
    <ArtworkItem artwork={ artwork } />
  ))
  .add('이미지 보이기', () => (
    <ArtworkItem artwork={ artwork } showImage />
  ))
