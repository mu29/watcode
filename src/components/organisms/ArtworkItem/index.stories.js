import React from 'react'
import { storiesOf } from '@storybook/react'
import ArtworkItem from '.'

const artwork = {
  code: 1120470,
  artist: 'nagatsuki taffee',
  title: 'Re: 제로부터 시작하는 이세계 생활',
  type: 'Light Novel',
  tags: ['rebirth', 'loop', 'harem', 'this is a long tag'],
  imageUrl: 'https://laftelimage.blob.core.windows.net/items/thumbs/large/395db1d0-ead8-407f-a316-04aee1e668a7.jpg',
}

storiesOf('Organisms/ArtworkItem', module)
  .add('기본', () => (
    <ArtworkItem artwork={ artwork } />
  ))
