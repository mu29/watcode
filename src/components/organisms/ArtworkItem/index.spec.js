import React from 'react'
import ArtworkItem from '.'

const artwork = {
  code: 112047,
  artist: 'nagatsuki taffee',
  title: 'Re: 제로부터 시작하는 이세계 생활',
  tags: ['rebirth', 'loop', 'harem', 'this is a long tag'],
  imageUrl: 'https://laftelimage.blob.core.windows.net/items/thumbs/large/395db1d0-ead8-407f-a316-04aee1e668a7.jpg',
}

const component = (props = {}) => withTheme(<ArtworkItem artwork={ artwork } { ...props } />)

describe('<ArtworkItem />', () => {
  it('정상적으로 렌더링된다', () => {
    const wrapper = render(component())
    expect(wrapper).toMatchSnapshot()
  })
})
