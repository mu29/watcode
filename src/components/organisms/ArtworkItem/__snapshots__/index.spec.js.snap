// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArtworkItem /> 정상적으로 렌더링된다 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0.75rem;
  padding-bottom: 0.5rem;
  background-color: #FFF;
  border-bottom: 0.0625rem solid #EEEEEE;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: 6rem;
  overflow: hidden;
}

.c3 {
  height: 3.2rem;
  overflow: scroll;
  margin-top: 0.375rem;
}

.c3::-webkit-scrollbar {
  width: 0;
  height: 0;
}

@media (max-width:767px) {
  .c0 {
    border-bottom: 0.0625rem solid #E0E0E0;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <span
      componentname="ArtworkImage"
    />
    <div
      class="c2"
    >
      <span
        componentname="Heading"
      >
        Re: 제로부터 시작하는 이세계 생활
      </span>
      <span
        componentname="Text"
      >
        by nagatsuki taffee
      </span>
      <div
        class="c3"
      >
        <span
          componentname="Tag"
        />
        <span
          componentname="Tag"
        >
          rebirth
        </span>
        <span
          componentname="Tag"
        >
          loop
        </span>
        <span
          componentname="Tag"
        >
          harem
        </span>
        <span
          componentname="Tag"
        >
          this is a long tag
        </span>
      </div>
    </div>
  </div>
  <span
    componentname="ArtworkReactions"
  />
</div>
`;
